test/functional/cluster/autopipelining.ts

```diff
@@ -177,6 +185,40 @@ describe("autoPipelining for cluster", function () {
     cluster.disconnect();
   });
 
+  it("should support building pipelines when a prefix is used", async () => {
+    const cluster = new Cluster(hosts, {
+      enableAutoPipelining: true,
+      keyPrefix: "baz:",
+    });
+    await new Promise((resolve) => cluster.once("connect", resolve));
+
+    await cluster.set("foo1", "bar1");
+    await cluster.set("foo10", "bar10");
+
+    expect(
+      await Promise.all([cluster.get("foo1"), cluster.get("foo10")])
+    ).to.eql(["bar1", "bar10"]);
+
+    cluster.disconnect();
+  });
+
+  it("should support building pipelines when a prefix is used with arrays to flatten", async () => {
+    const cluster = new Cluster(hosts, {
+      enableAutoPipelining: true,
+      keyPrefix: "baz:",
+    });
+    await new Promise((resolve) => cluster.once("connect", resolve));
+
+    await cluster.set(["foo1"], "bar1");
```

Comment:
This also fixes the autopipelining of commands which include arrays. Previously, args[0] was passed to the sharding function, where ioredis should have passed a string to the sharding function.

Annoyingly, `args[0]` is technically allowed to be the empty array or array-like by lodash, so this PR flattens the entire args list to account for that

```
> const { flatten } = require('./built/utils/lodash');
undefined
> flatten([[], [], ['key']])
[ 'key' ]
```